### PR TITLE
Bug fix: Trigger failure of hanging reindexes

### PR DESCRIPTION
**Pull request by annebyrne**
Previously, if a reindex would stall or hang without reindexing documents, then the job would just continue to persist in the `RunningJobs` table. With the updated ability to trigger bulk reindexes dependant on an empty running jobs queue, bulk reindexes would never trigger if a previous job had stalled.

This fix includes a check for whether the job has ran for a certain period without indexing documents. It also fixes an introduced bug in the `BulkActor` where completed jobs were determined by whether they were the latest job in the `JobHistory` table. If a manual reindex was ran immediately beforehand for the same content source & environment, then the job will always be considered to be completed.

Tested on `CODE`

### CODE REVIEW

File: app/com/gu/floodgate/reindex/BulkJobActor.scala [hunk 1]
```diff
@@ -162,17 +163,18 @@ class BulkJobActor(contentSources: List[ContentSource], runningJobService: Runni
...
-    val result = Future.sequence(completedSources.map { source =>
+    val completedAndRunningSources = orderedContentSources.filterNot(source => pendingReindexes.contains(source))
+
+    val result = Future.sequence(completedAndRunningSources.map { source =>
       jobHistoryService.getLatestJob(source.id, source.environment).map { maybeJobHistory =>
-        maybeJobHistory.map { jobHistory =>
-          CompletedJobInfo(source.appName, source.id, source.environment, jobHistory.startTime, jobHistory.finishTime, jobHistory.status)
+        maybeJobHistory.collect {
+          case jh if runningJobService.getRunningJob(jh.contentSourceId, jh.environment).isLeft =>
+            CompletedJobInfo(source.appName, source.id, source.environment, jh.startTime, jh.finishTime, jh.status)
```
Comment (LATaylor-guardian) [inline app/com/gu/floodgate/reindex/BulkJobActor.scala]: `CompletedJobInfo` could have an apply method on it's companion object which takes a `ContentSource` and `JobHistory` object in order to create a `CompletedJobInfo`.

File: app/com/gu/floodgate/reindex/ProgressTracker.scala [hunk 2]
```diff
@@ -106,9 +107,14 @@ class ProgressTracker(ws: WSClient, runningJobService: RunningJobService, jobHis
 
       case InProgress =>
         val runningJobUpdate = RunningJob(runningJob.contentSourceId, runningJob.contentSourceEnvironment, progress.documentsIndexed, progress.documentsExpected, runningJob.startTime, runningJob.rangeFrom, runningJob.rangeTo)
-        runningJobService.updateRunningJob(runningJob.contentSourceId, runningJob.contentSourceEnvironment, runningJobUpdate)
-        scheduleNextUpdate(contentSource, runningJob)
+        val periodIndexing = Minutes.minutesBetween(runningJobUpdate.startTime, DateTime.now())
 
+        if (runningJobUpdate.documentsIndexed == 0 && (periodIndexing.getMinutes - 10) >= 0) {
```
Comment (LATaylor-guardian) [inline app/com/gu/floodgate/reindex/ProgressTracker.scala]: You could assign the predicate here to a clearly named val for readability here.

File: public/components/bulkReindexController.react.js [hunk 3]
```diff
@@ -37,7 +37,7 @@ export default class BulkReindexControllerComponent extends React.Component {
     }
 
     componentDidMount() {
-        let intervalPeriod = 1000;
+        let intervalPeriod = 5000;
```
Comment (LATaylor-guardian) [inline public/components/bulkReindexController.react.js]: Why did you decide to extend this?

File: app/com/gu/floodgate/reindex/BulkJobActor.scala [hunk 4]
```diff
@@ -183,7 +185,7 @@ class BulkJobActor(contentSources: List[ContentSource], runningJobService: Runni
       case source :: xs => {
         reindexService.reindex(source.id, source.environment, new DateParameters(None, None)).map {
           case Left(error) => self ! ReindexError(source.id, source.environment, error)
-          case Right(runningJob) => self ! ReindexStarted(source.id, source.environment)
+          case Right(runningJob) => self ! ReindexStarted(runningJob.contentSourceId, runningJob.contentSourceEnvironment)
```
Comment (regiskuckaertz) [inline app/com/gu/floodgate/reindex/BulkJobActor.scala]: good catch 👍

File: public/components/bulkReindexController.react.js (hunk 3 above)
Comment (annebyrne) [reply public/components/bulkReindexController.react.js]: To reduce the amount of calls to the DynamoDB table. It didn't seem necessary to be calling every second.

File: app/com/gu/floodgate/reindex/BulkJobActor.scala (hunk 1 above)
Comment (annebyrne) [reply app/com/gu/floodgate/reindex/BulkJobActor.scala]: Good idea!

File: app/com/gu/floodgate/reindex/ProgressTracker.scala (hunk 2 above)
Comment (regiskuckaertz) [reply app/com/gu/floodgate/reindex/ProgressTracker.scala]: Where is `10` coming from?

Comment (regiskuckaertz) [reply app/com/gu/floodgate/reindex/ProgressTracker.scala]: I wonder if `runningJobUpdate` should be defined inside the `if` clause?

Comment (annebyrne) [reply app/com/gu/floodgate/reindex/ProgressTracker.scala]: Just an arbitrary choice - I looked at history of previous reindexes and ran a few in `CODE` and from that it looks as if all would have triggered reindex of one document by then.

File: public/components/bulkReindexController.react.js (hunk 3 above)
Comment (regiskuckaertz) [reply public/components/bulkReindexController.react.js]: Note for later: this is typically a property passed to the component and defined upstream.

**Pull request closed without merging**